tests/integration/routes/test_errors.py

```diff
@@ -80,3 +80,250 @@ def test_errors_500_exception_during_error_handling(self):
+        self.assertInUrl("/questionnaire/introduction/")
+
+        # When
+        current_url = self.last_url
+        self.exit()
+        self.get(current_url)
+
+        # Then
+        self.assertStatusUnauthorised()
+        cookie = self.getCookie()
+        self.assertEqual(cookie.get("theme"), "default")
+        self.assertInBody(
+            '<p>You will need to <a href="http://upstream.url/sign-in/logout">sign back in</a> to access your account</p>'
+        )
+
+    def test_401_theme_social_cookie_exists(self):
+        # Given
+        self.launchSurvey("test_theme_social")
+        self.assertInUrl("/questionnaire/radio/")
+
+        # When
+        current_url = self.last_url
+        self.saveAndSignOut()
+        self.get(current_url)
+
+        # Then
+        self.assertStatusUnauthorised()
+        cookie = self.getCookie()
+        self.assertEqual(cookie.get("theme"), "social")
+        self.assertInBody(
+            '<p>To access this page you need to <a href="https://rh.ons.gov.uk/sign-in/logout">re-enter your access code</a>.</p>'
+        )
+
+    def test_401_no_cookie(self):
+        # Given
+        self.launchSurvey("test_introduction")
+        self.assertInUrl("/questionnaire/introduction/")
+
+        # When
+        current_url = self.last_url
+        self.exit()
+        self.deleteCookieAndGetUrl(current_url)
+
+        # Then
+        self.assertStatusUnauthorised()
+        self.assertInBody(
+            '<p>If you are completing a business survey, you need to sign back in to <a href="https://surveys.ons.gov.uk/sign-in/logout">your account</a>.</p>'
+        )
+
+    def test_403_theme_default_cookie_exists(self):
+        # Given
+        self.launchSurvey("test_introduction")
+
+        # When
+        cookie = self.getUrlAndCookie("/dump/debug")
+
+        # Then
+        self.assertEqual(cookie.get("theme"), "default")
+        self.assertStatusForbidden()
+        self.assertInBody(
+            '<p>For further help, please <a href="http://upstream.url/contact-us/">contact us</a>.</p>'
+        )
+
+    def test_403_theme_social_cookie_exists(self):
+        # Given
+        self.launchSurvey("test_theme_social")
+
+        # When
+        cookie = self.getUrlAndCookie("/dump/debug")
+
+        # Then
+        self.assertEqual(cookie.get("theme"), "social")
+        self.assertStatusForbidden()
+        self.assertInBody(
+            '<p>For further help, please <a href="https://rh.ons.gov.uk/contact-us/">contact us</a>.</p>'
+        )
+
+    def test_403_no_cookie(self):
+        # Given
+        self.launchSurvey("test_introduction")
+
+        # When
+        token = 123
+        self.get(url=f"/session?token={token}")
+
+        # Then
+        self.assertStatusForbidden()
+        self.assertInBody(
+            '<p>If you are completing a business survey and you need further help, please <a href="https://surveys.ons.gov.uk/contact-us/">contact us</a>.</p>'
+        )
```

Comment:
Should assert both cases, so business and the "other". Same applies to the rest.

Reply:
Added  in templates where the "other" option exists, thanks!